api/src/controllers/note.ts

```diff
@@ -0,0 +1,130 @@
+import { Request, Response, NextFunction } from 'express';
+import noteDataController from '../data-controllers/note';
+
+const getNote = async (req: Request, res: Response, next: NextFunction) => {
+	const { userId } = res.locals.auth;
+	const { boardId, listId } = req.params;
+	const { noteId } = req.query;
+
+	if (!boardId) return res.status(400).json({ message: 'boardId was null' });
+	if (!listId) return res.status(400).json({ message: 'listId was null' });
+	if (!noteId || !(typeof noteId === 'string'))
+		return res.status(400).json({ message: 'noteId was null' });
+
+	const { note, err } = await noteDataController
+		.getNote(userId, boardId, listId, noteId)
+		.then(
+			(note) => ({ note, err: undefined }),
+			(err) => ({ err, note: undefined })
+		);
+	if (err) return next(err);
+	if (!note) return res.status(404).json({ message: 'Note not found' });
+
+	res.json({
+		note,
+	});
+};
+
+const updateNote = async (req: Request, res: Response, next: NextFunction) => {
+	const { userId } = res.locals.auth;
+	const { boardId, listId, noteId } = req.params;
+	const { name, description, startDate, dueDate, tagIds } = req.body;
+
+	if (!boardId) return res.status(400).json({ message: 'boardId was null' });
```

Comment:
same validation code as for GET/POST? Don't you want to generalize this?

Reply:
Example: once the param name is been changed, you will have to modify all the places where it's in use, instead a single update